src/llms/llm_loader.py

```diff
@@ -0,0 +1,233 @@
+# workaround to disable UserWarning
+import warnings
+warnings.simplefilter("ignore", UserWarning)
+
+from langchain.callbacks.streaming_stdout import StreamingStdOutCallbackHandler
+from langchain.callbacks.manager import CallbackManager
+
+import os, inspect
+
+from utils.logger import Logger
+
+class LLMLoader:
+    """
+    Note: This class loads the LLM backend libraries if the specific LLM is loaded.
+    Known caveats: Currently supports a single instance/model per backend
+
+    The llm_backends parameter is an array with the backend llm and (optional) parameters to override
+
+    # format for dictionary
+    llm_backend = {
+        'backend': 'openai',
+        'params': {...}     # optional key
+    }
+
+    # overriding specific parameters
+    llm_backend = {
+        'backend' : 'ollama',
+        'params': {'temperature': 0.02, 'top_p': 0.95},
+    }
+
+    # using default parameters
+    llm_backend = {'backend': 'tgi'}
+
+    # Using the class
+    llm_config = LLMLoader(llm_backend=llm_backend)
+    llm_chain = LLMChain(llm=llm_config.llm, prompt=prompt)
+
+    """
+    def __init__(self, llm_backend={}, logger=None) -> None:
```

Comment:
Don't set mutables (dict, list, ...) as defaults in init. https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments

Reply:
I actually had it as `llm_backend = None : dict` but reverted back last minute. Would `None` or `None : dict` preferred?

Reply:
I personally prefer `None: dict` which specifies both value and type (i.e more readability). But in case we are doing it lets do it for every function.

Reply:
`llm_backend: dict = None` should be fine

Reply:
Okay, I'll update with this

Reply:
updated